addons/l10n_in/models/account_invoice.py

```diff
@@ -71,23 +71,47 @@ def _compute_l10n_in_state_id(self):
 
     @api.depends('l10n_in_state_id')
     def _compute_fiscal_position_id(self):
-        other_country = None
-        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_sale_document(include_receipts=True))
+        foreign_state = False
+        in_moves = self.filtered(lambda move: move.country_code == 'IN' and move.is_invoice(include_receipts=True))
+        computed_ids = []
         for move in in_moves:
-            state_id = move.l10n_in_state_id.id
-            country_id = move.l10n_in_state_id.country_id.id
-            if move.l10n_in_state_id.l10n_in_tin == '96':  # Foreign Country
-                other_country = other_country or self.env['res.country'].search([('code', '!=', 'IN')], limit=1)
-                country_id = other_country.id
-                state_id = False
-            virtual_partner = self.env['res.partner'].new({
-                'state_id': state_id,
-                'country_id': country_id,
-            })
-            move.fiscal_position_id = self.env['account.fiscal.position'].with_company(
-                move.company_id
-            )._get_fiscal_position(virtual_partner)
-        super(AccountMove, self - in_moves)._compute_fiscal_position_id()
+            dummy_state_id = False
+            if move.l10n_in_state_id.l10n_in_tin == '96': # Foreign Country
+                if not foreign_state:
+                    foreign_state = self.env['res.country.state'].search([
+                        ('country_id', '!=', move.company_id.country_id.id)
+                    ], limit=1)
+                dummy_state_id = foreign_state
+
+            # In sale place of supply compare with company state
+            elif (
+                move.is_sale_document(include_receipts=True)
+                and move.partner_id.state_id != move.l10n_in_state_id
+            ):
+                dummy_state_id = move.l10n_in_state_id
+
+            # In purchase place of supply compare with vendor state
+            elif (
+                move.is_purchase_document(include_receipts=True)
+                and move.partner_id.country_id.code == 'IN'
+            ):
+                pos_state_id = move.l10n_in_state_id
+                # Vendor's place of supply is within his home state and not same as the company state
+                if pos_state_id == move.partner_id.state_id != move.company_id.state_id:
```

Comment:
```suggestion
                if pos_state_id == move.delivery_partner_id.state_id != move.company_id.state_id:
```
?

Reply:
`delivery_partner_id` is not for Bills (In Invoice)